.cargo/config

```diff
@@ -11,5 +11,18 @@ xclippy = [
 xlint = "run --package x --bin x -- lint"
 xtest = "run --package x --bin x -- external-crates-tests"
 
+# Configuration specifically for running clippy on `external-crates/move/`.
+# Some of these allows are to avoid code churn; others are filed as issues on the `sui` repo now.
+move-clippy = [
+    "clippy", "--",
+    "-Wclippy::all",
+    "-Wclippy::disallowed_methods",
+    "-Aclippy::upper_case_acronyms",
+    "-Aclippy::type_complexity",
+    "-Aclippy::new_without_default",
+    "-Aclippy::box_default",
+    "-Aclippy::manual_slice_size_calculation",
```

Comment:
@bmwill, thoughts on this config?

Reply:
Note a few of these are just because the code touches would be far-reaching (like renaming upper case acronyms) or because I felt they would be better-handled by the code's owner (such as removing manual slice size calculations from the VM internals).

Reply:
Some linting is better than no linting so this is fine to get things started. If we want we can chip away at these in subsequent PRs